cmd/queue/main.go

```diff
@@ -215,18 +215,21 @@ func handleKnativeProbe(w http.ResponseWriter, r *http.Request, ph string, probe
 		return
 	}
 
-	if !prober() {
-		http.Error(w, "container not ready", http.StatusServiceUnavailable)
-		probeSpan.Annotate([]trace.Attribute{
-			trace.StringAttribute("queueproxy.probe.error", "container not ready")}, "error")
-		return
-	}
-
-	// Respond with the name of the component handling the request.
-	w.Write([]byte(queue.Name))
+	healthState.HandleHealthProbe(func() bool {
+		if ready := prober(); !ready {
+			probeSpan.Annotate([]trace.Attribute{
+				trace.StringAttribute("queueproxy.probe.error", "container not ready")}, "error")
+			return false
+		}
+		return true
+	}, isAggressive, w, r)
 }
 
 func probeQueueHealthPath(port int, timeoutSeconds int) error {
+	if port <= 0 {
+		return errors.New("-port flag must be set a positive value")
```

Comment:
should this fatal rather? Or verified at startup?

Reply:
Any error from this func will result in existing with 1. I don't think we need to handle this error separately at startup (assuming you mean `main` func by startup).